src/components/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+import { deleteCompletedTodos } from '../api/todos';
+import { useTodosContext } from './useTodosContext';
+
+type Props = {
+  query: Status;
+  setQuery: (query: Status) => void;
+};
+
+export const Footer: React.FC<Props> = ({ query, setQuery }) => {
+  const {
+    todos,
+    setTodos,
+    setLoadingTodosIds,
+    handleError,
+    setIsInputFocused,
+  } = useTodosContext();
+  const completedIds = todos
+    .filter(item => item.completed === true)
+    .map(item => item.id);
+  const notCompletedQuantity = todos.filter(
+    (todo: Todo) => !todo.completed,
+  ).length;
+
+  const handleClearCompleted = () => {
+    setLoadingTodosIds(currentIds => [...currentIds, ...completedIds]);
+
+    deleteCompletedTodos(completedIds)
+      .then(() => {
+        setTodos(currentTodos => {
+          return currentTodos.filter(todo => !todo.completed);
+        });
+      })
+      .catch(() => handleError('Unable to delete completed todos'))
+      .finally(() => {
+        setLoadingTodosIds([]);
```

Comment:
You should remove only needed ids too, the same when you set them. Fix it in all places